### PR TITLE
Implement cybersource download.

### PR DESCRIPTION
**Pull request by brianhw**
Main code is feature complete.  Still needs tests, but would like review of approach.

@mulby @jab5569

### CODE REVIEW

**Comment by johnalbaker** [comment]
:+1:  Very nice!

File: edx/analytics/tasks/reports/payments.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+            'warehouse_path': self.warehouse_path,
+            'overwrite': self.overwrite,
+        }
+
+        current_date = start_date
+        while current_date < end_date:
+            args['run_date'] = current_date
+            task = DailyProcessFromCybersourceTask(**args)
+            # To cut down on the number of tasks that Luigi has to deal with,
+            # screen out here the tasks that are already complete.
```
Comment (mulby) [inline edx/analytics/tasks/reports/payments.py]: Is this necessary? I thought luigi could handle large volumes of tasks?

File: requirements/default.txt [hunk 2]
```diff
@@ -4,19 +4,28 @@ PyYAML==3.10		# MIT License
...
 httplib2==0.9 		# MIT
+idna==2.0               # BSD-like
 isoweek==1.3.0		# BSD
 mechanize==0.2.5	# BSD
 mysql-connector-python==1.2.2  	# GPL v2 with FOSS License Exception
+ndg-httpsclient==0.4.0  # BSD
 numpy==1.8.0 		# BSD
 pandas==0.13.0 		# BSD
 paramiko==1.14.0 	# LGPL
+paypalrestsdk==1.9.0    # Paypal SDK License
```
Comment (mulby) [inline requirements/default.txt:23]: Can we use this? I'm not sure what the terms of the Paypal SDK license are...

**Comment by mulby** [comment]
:+1: (comments are optional / don't need to be addressed yet)

**Comment by mulby** [comment]
As I'm working on the paypal capture, I'm wondering if we shouldn't name this file "cybersource.py" instead of the more generic "payments.py"

**Comment by mulby** [comment]
Also - I tend to prefer having an explicit output_root for low level tasks like this and only use the WarehouseMixin for tasks that are explicitly dealing with Hive tables etc. At that point they _need_ to put the data in the warehouse, but tasks upstream from there don't actually need to put it anywhere, so I tend to plumb the warehouse path into the output_root of those other tasks. It also means the lower level tasks don't need to know about how Hive organizes the data (at least the folder structure).

File: requirements/default.txt (hunk 2 above)
Comment (brianhw) [inline requirements/default.txt:23]: I had assumed that eCommerce folks had vetted it.  But I was wrong.  Pending...

File: edx/analytics/tasks/reports/payments.py (hunk 1 above)
Comment (mulby) [inline edx/analytics/tasks/reports/payments.py]: ping

**Pull request merged**